failing rpi tests

```
platform linux2 -- Python 2.7.9, pytest-3.0.6, py-1.4.32, pluggy-0.4.0 -- /home/pi/projects/pi-test/luma.examples/.tox/py27/bin/python2.7
cachedir: .cache
rootdir: /home/pi/projects/pi-test/luma.examples, inifile: 
plugins: cov-2.4.0
collected 12 items 

tests/test_demo_opts.py::test_support PASSED
tests/test_demo_opts.py::test_create_parser PASSED
tests/test_demo_opts.py::test_load_config_file_parse PASSED
tests/test_demo_opts.py::test_get_device_config_file_missing PASSED
tests/test_demo_opts.py::test_get_device_config__file_success PASSED
tests/test_demo_opts.py::test_get_choices_unknown_module PASSED
tests/test_demo_opts.py::test_get_device_unknown PASSED
tests/test_demo_opts.py::test_get_device_emulator_all PASSED
tests/test_demo_opts.py::test_get_device_led_matrix_all FAILED
tests/test_demo_opts.py::test_get_device_lcd_all FAILED
tests/test_demo_opts.py::test_get_device_oled_all FAILED
tests/test_demo_opts.py::test_get_device_oled_interface_unknown PASSED

================================================================ FAILURES ================================================================
_____________________________________________________ test_get_device_led_matrix_all _____________________________________________________

capsys = <_pytest.capture.CaptureFixture instance at 0x749794e0>

    def test_get_device_led_matrix_all(capsys):
        """
        Load supported led_matrix devices one by one.
        """
        for display in display_types.get('led_matrix'):
            with pytest.raises(SystemExit):
>               get_device(['--display', display])
E               Failed: DID NOT RAISE <type 'exceptions.SystemExit'>

tests/test_demo_opts.py:129: Failed
________________________________________________________ test_get_device_lcd_all _________________________________________________________

capsys = <_pytest.capture.CaptureFixture instance at 0x749798c8>

    def test_get_device_lcd_all(capsys):
        """
        Load supported lcd devices one by one.
        """
        for display in display_types.get('lcd'):
            with pytest.raises(SystemExit):
>               get_device(['--display', display])

tests/test_demo_opts.py:142: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
examples/demo_opts.py:150: in get_device
    device = Device(Serial(), **vars(args))
.tox/py27/local/lib/python2.7/site-packages/luma/lcd/device.py:64: in __init__
    self.contrast(0xB0)
.tox/py27/local/lib/python2.7/site-packages/luma/lcd/device.py:95: in contrast
    self.command(0x21, 0x14, value | 0x80, 0x20)
.tox/py27/local/lib/python2.7/site-packages/luma/core/device.py:39: in command
    self._serial_interface.command(*cmd)
.tox/py27/local/lib/python2.7/site-packages/luma/core/serial.py:84: in command
    list(cmd))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <smbus2.smbus2.SMBus object at 0x753ec4d0>, i2c_addr = 60, register = 0, data = [33, 20, 176, 32]

    def write_i2c_block_data(self, i2c_addr, register, data):
        # type: (int, int, list) -> None
        """
            Write a block of byte data to a given register
            :param i2c_addr: i2c address
            :param register: Start register
            :param data: List of bytes
            """
        length = len(data)
        if length > I2C_SMBUS_BLOCK_MAX:
            raise ValueError("Data length cannot exceed %d bytes" % I2C_SMBUS_BLOCK_MAX)
        self._set_address(i2c_addr)
        msg = i2c_smbus_ioctl_data.create(
            read_write=I2C_SMBUS_WRITE, command=register, size=I2C_SMBUS_I2C_BLOCK_DATA
        )
        msg.data.contents.block[0] = length
        msg.data.contents.block[1:length + 1] = data
>       ioctl(self.fd, I2C_SMBUS, msg)
E       IOError: [Errno 121] Remote I/O error

.tox/py27/local/lib/python2.7/site-packages/smbus2/smbus2.py:274: IOError
________________________________________________________ test_get_device_oled_all ________________________________________________________

capsys = <_pytest.capture.CaptureFixture instance at 0x73ec7b48>

    def test_get_device_oled_all(capsys):
        """
        Load supported oled devices one by one.
        """
        for display in display_types.get('oled'):
            with pytest.raises(SystemExit):
>               get_device(['--display', display])

tests/test_demo_opts.py:155: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
examples/demo_opts.py:142: in get_device
    device = Device(Serial(), **vars(args))
.tox/py27/local/lib/python2.7/site-packages/luma/oled/device.py:153: in __init__
    self._const.NORMALDISPLAY)
.tox/py27/local/lib/python2.7/site-packages/luma/core/device.py:39: in command
    self._serial_interface.command(*cmd)
.tox/py27/local/lib/python2.7/site-packages/luma/core/serial.py:84: in command
    list(cmd))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <smbus2.smbus2.SMBus object at 0x73efc390>, i2c_addr = 60, register = 0, data = [174, 213, 128, 168, 63, 211, ...]

    def write_i2c_block_data(self, i2c_addr, register, data):
        # type: (int, int, list) -> None
        """
            Write a block of byte data to a given register
            :param i2c_addr: i2c address
            :param register: Start register
            :param data: List of bytes
            """
        length = len(data)
        if length > I2C_SMBUS_BLOCK_MAX:
            raise ValueError("Data length cannot exceed %d bytes" % I2C_SMBUS_BLOCK_MAX)
        self._set_address(i2c_addr)
        msg = i2c_smbus_ioctl_data.create(
            read_write=I2C_SMBUS_WRITE, command=register, size=I2C_SMBUS_I2C_BLOCK_DATA
        )
        msg.data.contents.block[0] = length
        msg.data.contents.block[1:length + 1] = data
>       ioctl(self.fd, I2C_SMBUS, msg)
E       IOError: [Errno 121] Remote I/O error

.tox/py27/local/lib/python2.7/site-packages/smbus2/smbus2.py:274: IOError
=================================================== 3 failed, 9 passed in 5.75 seconds ===================================================
ERROR: InvocationError: '/home/pi/projects/pi-test/luma.examples/.tox/py27/bin/coverage run -m py.test -vv'
```

Results for python 3.5.3:

```
platform linux -- Python 3.5.3, pytest-3.0.6, py-1.4.32, pluggy-0.4.0 -- /home/pi/projects/pi-test/luma.examples/.tox/py35/bin/python3.5
cachedir: .cache
rootdir: /home/pi/projects/pi-test/luma.examples, inifile: 
plugins: cov-2.4.0
collected 12 items 

tests/test_demo_opts.py::test_support PASSED
tests/test_demo_opts.py::test_create_parser PASSED
tests/test_demo_opts.py::test_load_config_file_parse PASSED
tests/test_demo_opts.py::test_get_device_config_file_missing PASSED
tests/test_demo_opts.py::test_get_device_config__file_success PASSED
tests/test_demo_opts.py::test_get_choices_unknown_module PASSED
tests/test_demo_opts.py::test_get_device_unknown PASSED
tests/test_demo_opts.py::test_get_device_emulator_all PASSED
tests/test_demo_opts.py::test_get_device_led_matrix_all FAILED
tests/test_demo_opts.py::test_get_device_lcd_all FAILED
tests/test_demo_opts.py::test_get_device_oled_all FAILED
tests/test_demo_opts.py::test_get_device_oled_interface_unknown PASSED

================================================================ FAILURES ================================================================
_____________________________________________________ test_get_device_led_matrix_all _____________________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x7515df70>

    def test_get_device_led_matrix_all(capsys):
        """
        Load supported led_matrix devices one by one.
        """
        for display in display_types.get('led_matrix'):
            with pytest.raises(SystemExit):
>               get_device(['--display', display])
E               Failed: DID NOT RAISE <class 'SystemExit'>

tests/test_demo_opts.py:129: Failed
________________________________________________________ test_get_device_lcd_all _________________________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x73cebc10>

    def test_get_device_lcd_all(capsys):
        """
        Load supported lcd devices one by one.
        """
        for display in display_types.get('lcd'):
            with pytest.raises(SystemExit):
                get_device(['--display', display])
    
>           assertInError('error: GPIO access not available', capsys)

tests/test_demo_opts.py:144: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

msg = 'error: GPIO access not available', capsys = <_pytest.capture.CaptureFixture object at 0x73cebc10>

    def assertInError(msg, capsys):
        """
        Helper to find text in argparse error output message.
        """
        out, err = capsys.readouterr()
>       assert msg in err
E       assert 'error: GPIO access not available' in '/home/pi/projects/pi-test/luma.examples/.tox/py35/lib/python3.5/site-packages/luma/lcd/aux.py:28: RuntimeWarning: Thi...RATION] [--loop LOOP]\n               [--max-frames MAX_FRAMES]\ntest.py: error: I2C device not found on address: 60\n'

tests/test_demo_opts.py:35: AssertionError
________________________________________________________ test_get_device_oled_all ________________________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x751655d0>

    def test_get_device_oled_all(capsys):
        """
        Load supported oled devices one by one.
        """
        for display in display_types.get('oled'):
            with pytest.raises(SystemExit):
>               get_device(['--display', display])

tests/test_demo_opts.py:155: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
examples/demo_opts.py:142: in get_device
    device = Device(Serial(), **vars(args))
.tox/py35/lib/python3.5/site-packages/luma/oled/device.py:410: in __init__
    0xA4)               # Normal dislay
.tox/py35/lib/python3.5/site-packages/luma/core/device.py:39: in command
    self._serial_interface.command(*cmd)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <luma.core.serial.i2c object at 0x75cde0d0>, cmd = (174, 179, 242, 168, 63, 162, ...)

    def command(self, *cmd):
        """
            Sends a command or sequence of commands through to the I²C address
            - maximum allowed is 32 bytes in one go.
    
            :param cmd: a spread of commands
            :type cmd: int
            :raises luma.core.error.DeviceNotFoundError: I2C device could not be found.
            """
>       assert(len(cmd) <= 32)
E       AssertionError

.tox/py35/lib/python3.5/site-packages/luma/core/serial.py:80: AssertionError
=================================================== 3 failed, 9 passed in 5.19 seconds ===================================================
ERROR: InvocationError: '/home/pi/projects/pi-test/luma.examples/.tox/py35/bin/coverage run -m py.test -vv'
```

And another one:

```
_____________________________________________________ test_get_device_led_matrix_all _____________________________________________________

capsys = <_pytest.capture.CaptureFixture instance at 0x7491c440>

    def test_get_device_led_matrix_all(capsys):
        """
        Load supported led_matrix devices one by one.
        """
        for display in display_types.get('led_matrix'):
            try:
>               get_device(['--display', display])

tests/test_demo_opts.py:129: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
examples/demo_opts.py:157: in get_device
    device = Device(Serial(), **vars(args))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <luma.led_matrix.device.neopixel object at 0x74920ed0>, dma_interface = <luma.core.serial.spi object at 0x74920e70>, width = 128
height = 64, cascaded = None, rotate = 0, mapping = None
kwargs = {'bcm_backlight': 18, 'bcm_data_command': 24, 'bcm_reset': 25, 'bgr': False, ...}

    def __init__(self, dma_interface=None, width=8, height=4, cascaded=None,
                 rotate=0, mapping=None, **kwargs):
        super(neopixel, self).__init__(const=None, serial_interface=noop)
    
        # Derive (override) the width and height if a cascaded param supplied
        if cascaded is not None:
            width = cascaded
            height = 1
    
        self.cascaded = width * height
        self.capabilities(width, height, rotate, mode="RGB")
        self._mapping = list(mapping or range(self.cascaded))
        assert(self.cascaded == len(self._mapping))
        self._ws2812 = dma_interface or self.__ws2812__()
>       self._ws2812.init(width * height)
E       AttributeError: 'spi' object has no attribute 'init'

.tox/py27/local/lib/python2.7/site-packages/luma/led_matrix/device.py:197: AttributeError
```

--- a/tests/test_demo_opts.py
+++ b/tests/test_demo_opts.py
@@ -127,8 +127,8 @@ def test_get_device_led_matrix_all(capsys):
     for display in display_types.get('led_matrix'):
         try:
             get_device(['--display', display])
-        except SystemExit as e:
-            assert 'SPI device not found' in str(e)
+        except SystemExit:
+            assertInError('SPI device not found', capsys)
 
 
 # luma.lcd
@@ -138,11 +138,10 @@ def test_get_device_lcd_all(capsys):
     Load supported lcd devices one by one.
     """
     for display in display_types.get('lcd'):
-        with pytest.raises(SystemExit):
+        try:
             get_device(['--display', display])
-
-        assertInError('error: GPIO access not available', capsys)
-
+        except SystemExit:
+            assertInError('error: GPIO access not available', capsys)
 
 # luma.oled
 

--- a/tests/test_demo_opts.py
+++ b/tests/test_demo_opts.py
@@ -143,6 +143,7 @@ def test_get_device_lcd_all(capsys):
         except SystemExit:
             assertInError('error: GPIO access not available', capsys)
 
+
 # luma.oled
 
 def test_get_device_oled_all(capsys):

--- a/tests/test_demo_opts.py
+++ b/tests/test_demo_opts.py
@@ -144,7 +144,7 @@ def test_get_device_lcd_all(capsys):
             try:
                 assertInError('error: GPIO access not available', capsys)
             except:
-                assertInError('error: I2C device', capsys)
+                pass
 
 
 # luma.oled

--- a/tests/test_demo_opts.py
+++ b/tests/test_demo_opts.py
@@ -141,7 +141,10 @@ def test_get_device_lcd_all(capsys):
         try:
             get_device(['--display', display])
         except SystemExit:
-            assertInError('error: GPIO access not available', capsys)
+            try:
+                assertInError('error: GPIO access not available', capsys)
+            except:
+                assertInError('error: I2C device', capsys)
 
 
 # luma.oled

--- a/tests/test_demo_opts.py
+++ b/tests/test_demo_opts.py
@@ -125,10 +125,10 @@ def test_get_device_led_matrix_all(capsys):
     Load supported led_matrix devices one by one.
     """
     for display in display_types.get('led_matrix'):
-        with pytest.raises(SystemExit):
+        try:
             get_device(['--display', display])
-
-        assertInError('error: SPI device not found', capsys)
+        except SystemExit as e:
+            assert 'SPI device not found' in str(e)
 
 
 # luma.lcd

